bgls/utils.py

```diff
@@ -56,3 +61,157 @@ def cirq_density_matrix_bitstring_probability(
         density_matrix_state.target_tensor, num_qubits
     )
     return np.abs(density_matrix[index, index])
+
+
+def cirq_stabilizer_ch_bitstring_probability(
+    stabilizer_ch_form_state: cirq.sim.StabilizerChFormSimulationState,
+    bitstring: str,
+) -> float:
+    """Returns the probability of measuring the `bitstring` (|z⟩) in the
+    `cirq.StabilizerChFormSimulationState` (U_C U_H|s⟩), i.e. |⟨z|ψ⟩|^2.
+
+    Args:
+        stabilizer_ch_form_state: Stabilizer state in CH form (arxiv:
+        1808.00128) as a 'cirq.StabilizerChFormSimulationState'.
+        bitstring: Bitstring |z⟩ as a binary string.
+
+    """
+    # the state is of type StabilizerStateChForm
+    state = stabilizer_ch_form_state.state
+    index = int(bitstring, 2)
+    # this runs in O(n^2) for an n qubit state
+    return np.abs(state.inner_product_of_state_and_x(index)) ** 2
+
+
+def apply_near_clifford_gate(
```

Comment:
Suggest `act_on` or `act_on_stabilizer` - it's clearer and more consistent with Cirq.

Reply:
Updated this, sounds better